### PR TITLE
DOCUMENTATION: Add raylib wrapper documentation

### PR DESCRIPTION
**Pull request by kw4ntiq**
MINOR

* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] Norm of the code has been respected

* **In what state is your pull request?**
- [x] Ready to merge / Waiting a reviwer to see my work.
- [ ] Work In Progress (WIP) / My work is not finish but i want daily reviews. (Draft)
- [ ] CI Review / I want to see what the CI thinks of my work.
- [ ] Need feedback / Just to have feedback on what i produced. (May not be merged)

* **What kind of change does this PR introduce?** (You can choose multiple)
- [ ] Bug fix
- [ ] Feature request
- [x] New / Updated documentation
- [ ] Testing CI ( Make the pull request in draft mode)

* **What is the current behavior?** (link an issue based on the kind of change this pr introduce)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: docs/developer-guide/client/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+## Introduction
+The C++ client is a simple game client that uses the Raylib library to handle graphics and audio. The client is designed to be simple and easy to use, but it is also very flexible and can be used to create more complex games.
+
+## Raylib Library
+We use the Raylib library to handle our graphics and audio. Raylib is a simple and easy-to-use library that provides a wide range of features. The Raylib library is written in C, but we have wrapped it in C++ classes to make it easier to use.
+
+Explore the following sections to understand and utilize different aspects of the Raylib library:
+
+- [**Audio**](audio/index.html): Learn how to add sounds, music, and other audio effects to your game.
```
Comment (kw4ntiq) [inline docs/developer-guide/client/README.md]: I didn't manage to put audio/README.md, it directed me to the wrong thing. I checked the issues on mdbook's github that talked about this but I couldn't solve the problem.

File: docs/developer-guide/client/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  
+- [**Geometry**](inputs/index.html): Learn about the different data structures and how to use them.
+  
+- [**Events**](events/index.html): Detect and respond to application events like mouse clicks and keyboard inputs.
+  
+- [**Graphics**](graphics/index.html): Learn about advanced graphic rendering, music, animations...
```
Comment (Saverio976) [inline docs/developer-guide/client/README.md]: remove this, not needed

File: docs/developer-guide/client/audio/README.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+void stop() const;
+void pause() const;
+void resume() const;
+bool isPlaying() const;
+void setVolume(float volume) const;
+void setPitch(float pitch) const;
+void setPan(float pan) const;
+bool NeedToPlay() const;
+void setNeedToPlay(bool needToPlay);
+std::string getPath() const;
```
Comment (Saverio976) [inline docs/developer-guide/client/audio/README.md]: put Raylib::Sound:: before all method name

File: docs/developer-guide/client/audio/README.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+### Example usage
+
+```cpp
+Raylib::Sound mySound("path/to/soundfile.wav");
+mySound.play();
+mySound.setVolume(0.8f);
+```
+
+## Music Class
```
Comment (Saverio976) [inline docs/developer-guide/client/audio/README.md]: Put this in another file

File: docs/developer-guide/client/audio/README.md [hunk 5]
```diff
@@ -0,0 +1,124 @@
+# Using Audio in our game client
+
+In your C++ client application, you can manage audio using the Raylib library. This guide explains how to work with sounds and music using our Raylib wrapper classes.
+
+## Sound Class
```
Comment (Saverio976) [inline docs/developer-guide/client/audio/README.md]: Put this in another file

File: docs/developer-guide/client/events/README.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+  Check if a key is NOT being pressed.
+
+- `void setExitKey(KeyboardKey key);`
+  Set a key to exit the application.
+
+- `int getKeyPressed();`
+  Get the key pressed (keycode).
+
+- `int getCharPressed();`
+  Get the last character pressed (unicode).
```
Comment (Saverio976) [inline docs/developer-guide/client/events/README.md]: Raylib:: devant

File: docs/developer-guide/client/events/README.md [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+  Set the scaling factor for the mouse position.
+
+- `float getMouseWheelMove();`
+  Get the mouse wheel movement.
+
+- `Vector2 getMouseWheelMoveV();`
+  Get the mouse wheel movement as a vector.
+
+- `void setMouseCursor(int cursor);`
+  Set the mouse cursor style.
```
Comment (Saverio976) [inline docs/developer-guide/client/events/README.md]: Raylib::

File: docs/developer-guide/client/geometry/README.md [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+- `Rectangle(float x, float y, float width, float height);`
+  Constructor to create a rectangle with specified `x`, `y`, `width`, and `height` values.
+
+## Color Structure
+
+Represents a color.
+
+- `Color(uint8_t r, uint8_t g, uint8_t b, uint8_t a);`
+  Constructor to create a color with specified `r` (red), `g` (green), `b` (blue), and `a` (alpha) values.
```
Comment (Saverio976) [inline docs/developer-guide/client/geometry/README.md]: Raylib::

File: docs/developer-guide/client/graphic/README.md [hunk 9]
```diff
@@ -0,0 +1,308 @@
+# Using Graphics in our game client
+
+This guide explains how to work in your game client with images, sprites, text, ... using our Raylib wrapper classes.
+
+## Sprite Class
```
Comment (Saverio976) [inline docs/developer-guide/client/graphic/README.md]: in another file

File: docs/developer-guide/client/graphic/README.md [hunk 10]
```diff
@@ -0,0 +1,308 @@
...
+
+```cpp
+unsigned int getId() const;
+float getWidth() const;
+float getHeight() const;
+int getTextureWidth() const;
+int getTextureHeight() const;
+int getMipmaps() const;
+int getFormat() const;
+void unloadSprite();
```
Comment (Saverio976) [inline docs/developer-guide/client/graphic/README.md]: Raylib::Sprite

File: docs/developer-guide/client/graphic/README.md [hunk 11]
```diff
@@ -0,0 +1,308 @@
...
+### Example usage
+
+```cpp
+Raylib::Sprite mySprite("path/to/spritefile.png", 100.0f, 150.0f);
+float spriteWidth = mySprite.getWidth();
+std::cout << "Sprite width: " << spriteWidth << std::endl;
+mySprite.unloadSprite();
+```
+
+## Text Class
```
Comment (Saverio976) [inline docs/developer-guide/client/graphic/README.md]: put this in another file

File: docs/developer-guide/client/graphic/README.md [hunk 12]
```diff
@@ -0,0 +1,308 @@
...
+void drawEx(float spacing);
+void drawPro(Raylib::Vector2 origin, float rotation, float spacing);
+
+float x() const;
+float y() const;
+float getFontSize() const;
+void setFontSize(float fontSize);
+Raylib::Vector2 getPosition() const;
+void setPixelPosition(Raylib::Vector2 position);
+void setCurrentFontSize(float fontSize);
```
Comment (Saverio976) [inline docs/developer-guide/client/graphic/README.md]: Raylib::Text

File: docs/developer-guide/client/graphic/README.md [hunk 13]
```diff
@@ -0,0 +1,308 @@
...
+
+### Example usage
+
+```cpp
+Raylib::Text myText("Hello, World!", {100, 200}, 20.0f, Raylib::BLUE);
+myText.setPixelPosition({300, 400});
+myText.draw();
+```
+
+## Image Class
```
Comment (Saverio976) [inline docs/developer-guide/client/graphic/README.md]: put this in another file

File: docs/developer-guide/client/graphic/README.md [hunk 14]
```diff
@@ -0,0 +1,308 @@
...
+### Methods
+
+```cpp
+bool isImageReady() const;
+void unloadImage();
+int getWidth() const;
+int getHeight() const;
+int getMipmaps() const;
+int getFormat() const;
+void* getData() const;
```
Comment (Saverio976) [inline docs/developer-guide/client/graphic/README.md]: Raylib::Image::

File: docs/developer-guide/client/graphic/README.md [hunk 15]
```diff
@@ -0,0 +1,308 @@
...
+### Example usage
+
+```cpp
+Raylib::Image myImage("path/to/imagefile.png");
+int width = myImage.getWidth();
+std::cout << "Image width: " << width << std::endl;
+myImage.unloadImage();
+```
+
+## Shapes and drawing functions
```
Comment (Saverio976) [inline docs/developer-guide/client/graphic/README.md]: put this in another file

File: docs/developer-guide/client/graphic/README.md [hunk 16]
```diff
@@ -0,0 +1,308 @@
...
+
+```cpp
+void drawPixel(int posX, int posY, Raylib::Color color);
+void drawCircle(int centerX, int centerY, float radius, Raylib::Color color);
+void drawRectangle(int posX, int posY, int width, int height, Raylib::Color color);
```
Comment (Saverio976) [inline docs/developer-guide/client/graphic/README.md]: Raylib::

File: docs/developer-guide/client/graphic/README.md [hunk 17]
```diff
@@ -0,0 +1,308 @@
...
+
+### Example usage
+
+```cpp
+Raylib::drawPixel(100, 100, Raylib::RED);
+Raylib::drawCircle(200, 200, 50, Raylib::GREEN);
+Raylib::drawRectangle(300, 300, 80, 120, Raylib::BLUE);
+```
+
+## Window-related functions
```
Comment (Saverio976) [inline docs/developer-guide/client/graphic/README.md]: put this in another file

File: docs/developer-guide/client/graphic/README.md [hunk 18]
```diff
@@ -0,0 +1,308 @@
...
+  Set configuration flags for the window.
+
+- `bool isWindowResized();`
+  Check if the window has been resized.
+
+- `bool isWindowState(ConfigFlags flag);`
+  Check if a specific window state flag is set.
+
+- `void setWindowState(ConfigFlags flag);`
+  Set a specific window state flag.
```
Comment (Saverio976) [inline docs/developer-guide/client/graphic/README.md]: Raylib::

File: docs/developer-guide/client/graphic/README.md [hunk 19]
```diff
@@ -0,0 +1,308 @@
...
+  Get the current screen height.
+
+- `int getRenderWidth();`
+  Get the current rendering width
+
+- `int getRenderHeight();`
+  Get the current rendering height
+
+- `int getMonitorWidth(int monitor);`
+  Get the width of the specified monitor.
```
Comment (Saverio976) [inline docs/developer-guide/client/graphic/README.md]: Raylib::

File: docs/developer-guide/client/graphic/README.md [hunk 20]
```diff
@@ -0,0 +1,308 @@
...
+- `int getCurrentMonitor();`
+  Get the index of the current monitor.
+
+- `void setClipboardText(const std::string &text);`
+  Set the text to be copied to the clipboard.
+
+- `std::string getClipboardText();`
+  Get the text currently copied to the clipboard.
+
+- `void setWindowIcon(Image image);`
```
Comment (Saverio976) [inline docs/developer-guide/client/graphic/README.md]: Raylib::

File: docs/developer-guide/client/graphic/README.md [hunk 21]
```diff
@@ -0,0 +1,308 @@
...
+  Set the window icon.
+
+## Cursor-related functions
```
Comment (Saverio976) [inline docs/developer-guide/client/graphic/README.md]: put this in another file

File: docs/developer-guide/client/graphic/README.md [hunk 22]
```diff
@@ -0,0 +1,308 @@
...
+  Check if the cursor is currently hidden.
+
+- `void enableCursor();`
+  Enable cursor (unlock it).
+
+- `void disableCursor();`
+  Disable cursor (lock it).
+
+- `bool isCursorOnScreen();`
+  Check if the cursor is within the game window.
```
Comment (Saverio976) [inline docs/developer-guide/client/graphic/README.md]: Raylib::

File: docs/developer-guide/client/graphic/README.md [hunk 23]
```diff
@@ -0,0 +1,308 @@
...
+
+## Drawing-related functions
```
Comment (Saverio976) [inline docs/developer-guide/client/graphic/README.md]: put this in another file

File: docs/developer-guide/client/graphic/README.md [hunk 24]
```diff
@@ -0,0 +1,308 @@
...
+
+- `void clearBackground(Raylib::Color color);`
+  Clear the background with a specified color.
+
+- `void beginDrawing();`
+  Begin drawing.
+
+- `void endDrawing();`
+  End drawing and swap buffers.
```
Comment (Saverio976) [inline docs/developer-guide/client/graphic/README.md]: Raylib::

File: docs/developer-guide/client/graphic/README.md [hunk 25]
```diff
@@ -0,0 +1,308 @@
...
+
+## Timing-related functions
```
Comment (Saverio976) [inline docs/developer-guide/client/graphic/README.md]: put this in another file

File: docs/developer-guide/client/graphic/README.md [hunk 26]
```diff
@@ -0,0 +1,308 @@
...
+  Set the target frames-per-second.
+
+- `int getFPS();`
+  Get the current frames-per-second.
+
+- `float getFrameTime();`
+  Get the time in seconds for a frame.
+
+- `double getTime();`
+  Get the current time in seconds.
```
Comment (Saverio976) [inline docs/developer-guide/client/graphic/README.md]: Raylib::

File: docs/developer-guide/client/graphic/README.md [hunk 27]
```diff
@@ -0,0 +1,308 @@
...
+
+## Misc. functions
```
Comment (Saverio976) [inline docs/developer-guide/client/graphic/README.md]: put this in another file

File: docs/developer-guide/client/graphic/README.md [hunk 28]
```diff
@@ -0,0 +1,308 @@
...
+
+- `void takeScreenshot(const std::string &fileName);`
+  Take a screenshot and save it to a file with the specified name.
```
Comment (Saverio976) [inline docs/developer-guide/client/graphic/README.md]: Raylib::

File: docs/developer-guide/client/graphic/README.md [hunk 29]
```diff
@@ -0,0 +1,308 @@
...
+
+## Shapes-related functions
```
Comment (Saverio976) [inline docs/developer-guide/client/graphic/README.md]: put this in another file

File: docs/developer-guide/client/graphic/README.md [hunk 30]
```diff
@@ -0,0 +1,308 @@
...
+
+- `void drawPixel(int posX, int posY, Raylib::Color color);`
```
Comment (Saverio976) [inline docs/developer-guide/client/graphic/README.md]: Raylib::

File: docs/developer-guide/client/graphic/README.md [hunk 31]
```diff
@@ -0,0 +1,308 @@
...
+  Draw a pixel at the specified position with the given color.
+
+- `void drawCircle(int centerX, int centerY, float radius, Raylib::Color color);`
+  Draw a circle with the specified center, radius, and color.
+
+- `void drawRectangle(int posX, int posY, int width, int height, Raylib::Color color);`
+  Draw a rectangle at the specified position with the given width, height, and color.
```
Comment (Saverio976) [inline docs/developer-guide/client/graphic/README.md]: Raylib::

File: docs/developer-guide/client/graphic/README.md [hunk 32]
```diff
@@ -0,0 +1,308 @@
...
+
+## Color/pixel related functions
```
Comment (Saverio976) [inline docs/developer-guide/client/graphic/README.md]: put this in another file

File: docs/developer-guide/client/graphic/README.md [hunk 33]
```diff
@@ -0,0 +1,308 @@
...
+  Convert a Color to a 32-bit integer.
+
+- `Vector4 colorNormalize(Raylib::Color color);`
+  Normalize a Color struct to a Vector4.
+
+- `Raylib::Color colorFromNormalized(Vector4 normalized);`
+  Create a Color from a normalized Vector4.
+
+- `Raylib::Color getColor(unsigned int hexValue);`
+  Create a Color from a hex value.
```
Comment (Saverio976) [inline docs/developer-guide/client/graphic/README.md]: Raylib::

File: docs/developer-guide/client/graphic/README.md [hunk 34]
```diff
@@ -0,0 +1,308 @@
...
+
+## Enum ConfigFlags
```
Comment (Saverio976) [inline docs/developer-guide/client/graphic/README.md]: put this in another file

File: docs/developer-guide/client/graphic/README.md [hunk 35]
```diff
@@ -0,0 +1,308 @@
...
+- `FLAG_WINDOW_TRANSPARENT`: Allow the window to be transparent.
+- `FLAG_WINDOW_HIDDEN`: Hide the window.
+- `FLAG_WINDOW_MINIMIZED`: Minimize the window.
+- `FLAG_WINDOW_MAXIMIZED`: Maximize the window.
+- `FLAG_WINDOW_UNFOCUSED`: Disable window focus.
+- `FLAG_WINDOW_TOPMOST`: Set the window to be always on top.
+- `FLAG_WINDOW_HIGHDPI`: Enable high-DPI mode.
+- `FLAG_WINDOW_ALWAYS_RUN`: Allow the window to run in the background.
+- `FLAG_MSAA_4X_HINT`: Enable 4x MSAA.
+- `FLAG_VSYNC_HINT`: Enable V-Sync.
```
Comment (Saverio976) [inline docs/developer-guide/client/graphic/README.md]: Raylib::

File: docs/developer-guide/client/graphic/README.md [hunk 36]
```diff
@@ -0,0 +1,308 @@
...
+- `Raylib::Color colorFromNormalized(Vector4 normalized);`
+  Create a Color from a normalized Vector4.
+
+- `Raylib::Color getColor(unsigned int hexValue);`
+  Create a Color from a hex value.
+
+## Enum ConfigFlags
+
+You can use the following flags to configure the window:
+
```
Comment (Saverio976) [inline docs/developer-guide/client/graphic/README.md]: example of usage ?

**Review by Saverio976 — APPROVED** [review]

**Review by Saverio976 — CHANGES_REQUESTED** [review]
comment

File: docs/developer-guide/client/graphic/README.md (hunk 35 above)
Comment (kw4ntiq) [reply docs/developer-guide/client/graphic/README.md]: @Saverio976 c pas dans raylib c dans Raylib::ConfigFlags::

**Comment by kw4ntiq** [comment]
> comment

comment of what

File: docs/developer-guide/client/events/keyboard.md [hunk 37]
```diff
@@ -0,0 +1,22 @@
+## Input-related functions: keyboard
+
+- `bool Raylib::isKeyPressed(KeyboardKey key);`
+  Check if a key has been pressed once.
+
+- `bool Raylib::isKeyDown(KeyboardKey key);`
```
Comment (Saverio976) [inline docs/developer-guide/client/events/keyboard.md]: Raylib::

File: docs/developer-guide/client/events/mouse.md [hunk 38]
```diff
@@ -0,0 +1,56 @@
+## Input-related functions: mouse
+
+- `bool Raylib::isMouseButtonPressed(MouseButton button);`
```
Comment (Saverio976) [inline docs/developer-guide/client/events/mouse.md]: Raylib::

File: docs/developer-guide/client/geometry/constant.md [hunk 39]
```diff
@@ -0,0 +1,26 @@
+### Color Constants
```
Comment (Saverio976) [inline docs/developer-guide/client/geometry/constant.md]: Et pourquoi pas dans le fichier color ?

File: docs/developer-guide/client/geometry/vector2.md [hunk 40]
```diff
@@ -0,0 +1,6 @@
+## Vector2 Structure
+
+Represents a 2D vector.
+
+- `Raylib::Vector2(float x, float y);`
+  Constructor to create a 2D vector with specified `x` and `y` values.
```
Comment (Saverio976) [inline docs/developer-guide/client/geometry/vector2.md]: \n

File: docs/developer-guide/client/geometry/vector4.md [hunk 41]
```diff
@@ -0,0 +1,6 @@
+## Vector4 Structure
```
Comment (Saverio976) [inline docs/developer-guide/client/geometry/vector4.md:1]: Diff between vector4 et rectangle

File: docs/developer-guide/client/graphic/colorsPixels.md [hunk 42]
```diff
@@ -0,0 +1,16 @@
...
+- `Raylib::Color Raylib::fade(Raylib::Color color, float alpha);`
+  Fade a color by a specified alpha value.
+
+- `int Raylib::colorToInt(Raylib::Color color);`
+  Convert a Color to a 32-bit integer.
+
+- `Vector4 Raylib::colorNormalize(Raylib::Color color);`
+  Normalize a Color struct to a Vector4.
+
+- `Raylib::Color Raylib::colorFromNormalized(Vector4 normalized);`
```
Comment (Saverio976) [inline docs/developer-guide/client/graphic/colorsPixels.md]: Raylib::

File: docs/developer-guide/client/graphic/configFlags.md [hunk 43]
```diff
@@ -0,0 +1,23 @@
...
+- `Raylib::ConfigFlags::FLAG_WINDOW_HIGHDPI`: Enable high-DPI mode.
+- `Raylib::ConfigFlags::FLAG_WINDOW_ALWAYS_RUN`: Allow the window to run in the background.
+- `Raylib::ConfigFlags::FLAG_MSAA_4X_HINT`: Enable 4x MSAA.
+- `Raylib::ConfigFlags::FLAG_VSYNC_HINT`: Enable V-Sync.
+
+## Example
+
+```cpp
+Raylib::setWindowState(Raylib::ConfigFlags::WINDOW_RESIZABLE);
+```
```
Comment (Saverio976) [inline docs/developer-guide/client/graphic/configFlags.md:24]: How to enable full screen and vsync at same time example. ?

**Review by Saverio976 — CHANGES_REQUESTED** [review]

File: docs/developer-guide/client/graphic/configFlags.md (hunk 43 above)
Comment (kw4ntiq) [reply docs/developer-guide/client/graphic/configFlags.md:24]: its not a "how to code in cpp doc" its a raylib wrapper doc

File: docs/developer-guide/client/geometry/vector4.md (hunk 41 above)
Comment (kw4ntiq) [reply docs/developer-guide/client/geometry/vector4.md:1]: It's a raylib wrapper so it takes its functionalities.
For more information, see the rectangle struct (l.244) and the Vector4 struct(l.217) in the original raylib.h file at https://github.com/raysan5/raylib/blob/master/src/raylib.h.
The Vector4 struct represents a vector with four components (x, y, z, w), while the Rectangle struct represents a rectangle with position and size properties (x, y, width, height) in 2D.

For more more more informations about geometry:
https://en.wikipedia.org/wiki/Rectangle
https://docs.fivem.net/docs/scripting-reference/runtimes/lua/functions/vector4/

**Review by Saverio976 — APPROVED** [review]

**Pull request merged**